mmdet3d/core/bbox/coders/smoke_bbox_coder.py

```diff
@@ -0,0 +1,191 @@
+import numpy as np
+import torch
+
+from mmdet.core.bbox import BaseBBoxCoder
+from mmdet.core.bbox.builder import BBOX_CODERS
+
+
+@BBOX_CODERS.register_module()
+class SMOKECoder(BaseBBoxCoder):
+    """Bbox Coder for SMOKE.
+
+    Args:
+        base_depth (tuple[float]): Depth references for decode box depth.
+        base_dim (tuple[tuple[float]]): Dimension references for decode
+            box dimension for each category. fomat [l, w, h]
```

Comment:
Use base_dims to distinguish with the case of base_depth (tuple[tuple[float]] vs tuple[float])